pkg/controllers/common/constants.go

```diff
@@ -127,6 +127,9 @@ const (
 	PodUnschedulableThresholdAnnotation = InternalPrefix + "pod-unschedulable-threshold"
 	// AutoMigrationInfoAnnotation contains auto migration information.
 	AutoMigrationInfoAnnotation = DefaultPrefix + "auto-migration-info"
+	// SourcePruningMergePatchAnnotation indicates the pruning merge patch document capable of converting
```

Comment:
Should we remove "pruning" from the name of this annotation? In case we have to perform any non-pruning operations in the future?  Perhaps something like `template-generate-merge-patch`? cc @gary-lgy